firstvoices/backend/models/page.py

```diff
@@ -0,0 +1,65 @@
+from django.core.validators import validate_slug
+from django.db import models
+from django.db.models import Q
+from django.utils.translation import gettext as _
+
+from backend.models import Image
+from backend.models.base import BaseControlledSiteContentModel
+from backend.models.media import Video
+from backend.models.widget import SiteWidgetList
+from backend.permissions import predicates
+
+
+class SitePage(BaseControlledSiteContentModel):
+    class Meta:
+        verbose_name = _("site page")
+        verbose_name_plural = _("site pages")
+        rules_permissions = {
+            "view": predicates.is_visible_object,
+            "add": predicates.is_superadmin,  # permissions will change when we add a write API
+            "change": predicates.is_superadmin,
+            "delete": predicates.is_superadmin,
+        }
+        constraints = [
+            # Constraint to ensure a SitePage has either a banner image or video or none but not both.
```

Comment:
Is there a way to reuse this between SitePage and Site?

Reply:
It looks like constraints are not reusable as they require a unique name across models. I have updated the constraint to match the one in the site model though.